### PR TITLE
Add UserClaims.cs and requested methods for frontend 

### PR DESCRIPTION
**Pull request by ThomasITU**
Add UserClaims for creating users - We need to figure out how to post, or this won't matter 

Hookup requested methods needed for frontend

### CODE REVIEW

File: Server/Controllers/BasicController.cs [hunk 1]
```diff
@@ -75,50 +75,38 @@ public async Task<ActionResult<RatingDTO>> PatchRating(CreateRatingDTO rating)
     [HttpGet("{SearchString}")]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public Task<ActionResult<IReadOnlyCollection<MaterialDTO>>> Search(string searchInput)
+    public async Task<IReadOnlyCollection<MaterialDTO>> Search(string searchInput)
```
Comment (MaCoHa) [inline Server/Controllers/BasicController.cs:78]: der skal være ActionResult

File: Server/Controllers/BasicController.cs [hunk 2]
```diff
@@ -75,50 +75,38 @@ public async Task<ActionResult<RatingDTO>> PatchRating(CreateRatingDTO rating)
...
 
-
-   
-
     [Authorize(Roles = $"{Roles.Teacher},{Roles.Student},{Roles.Administrator},{Roles.User}")]
     [HttpGet("Recommended/{Id}")]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public Task<ActionResult<IReadOnlyCollection<MaterialDTO>>> FindRecommendedMaterials(string Id)
+    public async Task<IReadOnlyCollection<MaterialDTO>> FindRecommendedMaterials(int Id)
```
Comment (MaCoHa) [inline Server/Controllers/BasicController.cs:88]: ActionResult

File: Server/Controllers/BasicController.cs [hunk 3]
```diff
@@ -75,50 +75,38 @@ public async Task<ActionResult<RatingDTO>> PatchRating(CreateRatingDTO rating)
...
-        //Relatede Material Code HERE
-      
-    }
-
-    [Authorize]
-    [HttpPost]
-    [ProducesResponseType(StatusCodes.Status200OK)]
-    [ProducesResponseType(StatusCodes.Status404NotFound)]
-        public async Task<IActionResult> Post(List<string> CommentInfo)
+    public async Task<(Status status, IReadOnlyCollection<CommentDTO>? comments)> GetCommentMaterials(int materialId)
```
Comment (MaCoHa) [inline Server/Controllers/BasicController.cs:98]: ActionResult

File: Server/Controllers/BasicController.cs (hunk 2 above)
Comment (MaCoHa) [inline Server/Controllers/BasicController.cs:88]: ActionResult

File: Server/Controllers/BasicController.cs (hunk 1 above)
Comment (MaCoHa) [inline Server/Controllers/BasicController.cs:78]: ActionResult

**Review by MaCoHa — CHANGES_REQUESTED** [review]
Status skal skiftes ud med ActionResult

File: Infrastructure/Repositories/CommentRepository.cs [hunk 4]
```diff
@@ -24,13 +24,17 @@ public async Task<(Status status, CommentDTO comment)> CreateAsync(CreateComment
...
-                       select new CommentDTO(c.Id, c.MaterialId, c.UserId, c.Text);
-
-        return await comments.ToListAsync();
+                       select new CommentDTO(c.Id, c.MaterialId, c.UserId, c.Text))
+                       .ToListAsync();
+        if(comments.Count > 0)
+        {
+            return (Found, comments);
+        }
+        return (NotFound,null);
```
Comment (Rasmus-Balthazar) [inline Infrastructure/Repositories/CommentRepository.cs:37]: Should return an empty list

**Pull request closed without merging**